CS483/CS483/Kartaclysm/Components/Abilities/ComponentStrikeAbility.cpp

```diff
@@ -0,0 +1,129 @@
+//----------------------------------------------------------------------------
+// ComponentStrikeAbility.h
+// Author: Bradley Cooper
+//
+// King Pin's ability to fire a linear projectile.
+//----------------------------------------------------------------------------
+
+#include "ComponentStrikeAbility.h"
+
+namespace Kartaclysm
+{
+	ComponentStrikeAbility::ComponentStrikeAbility(
+		HeatStroke::GameObject* p_pGameObject,
+		std::string p_strProjectileXML)
+		:
+		ComponentAbility(p_pGameObject),
+		m_strProjectileXML(p_strProjectileXML)
+	{
+		// Listen to on hit event
+		m_pOnHitDelegate = new std::function<void(const HeatStroke::Event*)>(std::bind(&ComponentStrikeAbility::OnHitCallback, this, std::placeholders::_1));
+		HeatStroke::EventManager::Instance()->AddListener(m_strPlayerX + "_StrikeHit", m_pOnHitDelegate);
+	}
+
+	ComponentStrikeAbility::~ComponentStrikeAbility()
+	{
+		HeatStroke::EventManager::Instance()->RemoveListener(GetGameObject()->GetGUID(), m_pAbilityDelegate);
+		delete m_pAbilityDelegate;
+		m_pAbilityDelegate = nullptr;
+
+		HeatStroke::EventManager::Instance()->RemoveListener(m_strPlayerX + "_StrikeHit", m_pAbilityDelegate);
+		delete m_pOnHitDelegate;
+		m_pOnHitDelegate = nullptr;
+	}
+
+	HeatStroke::Component* ComponentStrikeAbility::CreateComponent(
+		HeatStroke::GameObject* p_pGameObject,
+		tinyxml2::XMLNode* p_pBaseNode,
+		tinyxml2::XMLNode* p_pOverrideNode)
+	{
+		assert(p_pGameObject != nullptr);
+
+		// Defaults
+		std::string strProjectileXML("");
+
+		// All parameters are optional.
+		if (p_pBaseNode != nullptr)
+		{
+			ParseNode(p_pBaseNode, strProjectileXML);
+		}
+		if (p_pOverrideNode != nullptr)
+		{
+			ParseNode(p_pOverrideNode, strProjectileXML);
+		}
+
+		// Check that we got everything we needed.
+		assert(strProjectileXML != "");
```

Comment:
Again you're saying it's optional, then asserting it's not the default.  Which is it?